homeassistant/components/emoncms/config_flow.py

```diff
@@ -0,0 +1,203 @@
+    return [
+        {
+            "value": feed[FEED_ID],
+            "label": f"{feed[FEED_ID]}|{feed[FEED_TAG]}|{feed[FEED_NAME]}",
+        }
+        for feed in feeds
+    ]
+
+
+async def get_feed_list(hass: HomeAssistant, url: str, api_key: str) -> dict[str, Any]:
+    """Check connection to emoncms and return feed list if successful."""
+    emoncms_client = EmoncmsClient(
+        url,
+        api_key,
+        session=async_get_clientsession(hass),
+    )
+    return await emoncms_client.async_request("/feed/list.json")
+
+
+class EmoncmsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """emoncms integration UI config flow."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlowWithConfigEntry:
+        """Get the options flow for this handler."""
+        return EmoncmsOptionsFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Initiate a flow via the UI."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {
+                    CONF_API_KEY: user_input[CONF_API_KEY],
+                    CONF_ID: user_input[CONF_ID],
+                    CONF_URL: user_input[CONF_URL],
+                }
+            )
+            result = await get_feed_list(
+                self.hass, user_input[CONF_URL], user_input[CONF_API_KEY]
+            )
+            if not result[CONF_SUCCESS]:
+                errors["base"] = result[CONF_MESSAGE]
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_ID], data=user_input
+                )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_ID): str,
+                        vol.Required(CONF_URL): str,
+                        vol.Required(CONF_API_KEY): str,
+                    }
+                ),
+                user_input,
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_info: ConfigType) -> ConfigFlowResult:
+        """Import config from yaml."""
+        url = import_info[CONF_URL]
+        api_key = import_info[CONF_API_KEY]
+        value_template = None
+        exclude_feeds = None
+        include_only_feeds = None
+        if import_info.get(CONF_VALUE_TEMPLATE) is not None:
+            value_template = import_info[CONF_VALUE_TEMPLATE].template
+        if import_info.get(CONF_EXCLUDE_FEEDID) is not None:
+            exclude_feeds = list(map(str, import_info[CONF_EXCLUDE_FEEDID]))
+        if import_info.get(CONF_ONLY_INCLUDE_FEEDID) is not None:
+            include_only_feeds = list(map(str, import_info[CONF_ONLY_INCLUDE_FEEDID]))
```

Comment:
So I think there are 2 things we need to figure out before we can continue:
1. The integration uses templates in YAML to pre process their data (if I understood correctly). We don't want that kind of feature when we move to the config flow. Integrations should expose contextualized "raw" data from the API or device. The user can then use templates and helpers and what not to tailer it to their needs.
2. Config flow integrations are one size fits all kind of deal. Where previously you could be picky with the YAML setup to only add entities which you wish. With config flows we want to keep it simple. The main config flow should just ask for the url and api key (like we do now) and just add everything. The user can then disable stuff as they wish. I think we should ignore the YAML config and just have 1 type of config entry (so we don't have 1 "new" config entry which adds everything and 1 "old" config entry which omits stuff).

Reply:
not really preprocessing, mostly to choose feeds they want to include and to exclude. When you have 100 feeds imported in one shot and you want to keep only 10, it is a lot of work for the user to remove the 90 feeds he does not want to keep. So I thought it more friendly not to import nothing at the entry creation, just to test that the connection and then when the user click on configure he chooses the feeds he wants to include....

I agree it is better to remove the `value_template`

Reply:
Can you give more context about how many feeds an average instance has?

Reply:
60 or 100 feeds is something usual. @borpin, can you confirm or give further details?

Reply:
@joostlek - there is no typical use. As a minimum, on an older system 6-10 available Feeds, at extremis 1000 (or more and everything in between). I have about 300 in one instance, but a lot of these are actually coming *from* HA (so would not want them back again). There also may be historical feeds that are no longer updated, but kept.

There is no limit on the number Feeds that can be created.

One node/tag
![image](https://github.com/user-attachments/assets/f424fc4a-7f59-4273-b7ee-8fd251ef4ddc)

@alexandercuer - Can you do the Feed selection by Node, then feed?

I would agree that you only want the user to pull in the Feeds they want, allowing the user to select the feeds required (as per the current design). A 1000 disabled feeds would be a pest in HA.

1. Reduce load on emoncms instance API - only pull required feeds
2. If a feed is misconfigured, you do not want to be loading it into HA.
3. Minimise the list of sensors available to the user.

I'd also say, try and keep the sensor name compact. I'd suggest <instance>-<node>-<feed> - (I've not kept up with what you are proposing), but some integrations have impossibly long sensor names.

One other thought - the FeedID is only ever used once and cannot be changed. The Name/description in emoncms can be changed - could this simply pass through to HA?

Reply:
Only kept `CONF_ONLY_INCLUDE_FEEDID` from the YAML config, so this should be close to what you want